packages/lib/modules/pool/actions/initialize/usePathToInitializePool.ts

```diff
@@ -0,0 +1,249 @@
+import { useEffect, useMemo } from 'react'
+import { usePoolCreationForm } from '../create/PoolCreationFormProvider'
+import { useParams } from 'next/navigation'
+import { useReadContracts, useReadContract } from 'wagmi'
+import { formatUnits, Address, erc20Abi, parseAbi } from 'viem'
+import { getChainId } from '@repo/lib/config/app.config'
+import { GqlChain } from '@repo/lib/shared/services/api/generated/graphql'
+import { AddressProvider } from '@balancer/sdk'
+import { vaultExtensionAbi_V3 } from '@balancer/sdk'
+import { PERCENTAGE_DECIMALS } from '../create/constants'
+import { usePoolCreationFormSteps } from '../create/usePoolCreationFormSteps'
+import { PoolCreationToken, SupportedPoolTypes } from '../create/types'
+import { WeightedPoolStructure } from '../create/constants'
+import { PoolType } from '@balancer/sdk'
+
+export function usePathToInitializePool() {
+  const { poolCreationForm, poolAddress, setPoolAddress, reClammConfigForm } = usePoolCreationForm()
+  const { lastStep } = usePoolCreationFormSteps()
+  const params = useParams()
+
+  const networkParam = params.network as GqlChain
+  const poolAddressParam = params.poolAddress as Address
+  const poolTypeParam = params.poolType as SupportedPoolTypes
+
+  const isStablePool = poolTypeParam === PoolType.Stable || poolTypeParam === PoolType.StableSurge
+  const isWeightedPool = poolTypeParam === PoolType.Weighted
+  const isReClammPool = poolTypeParam === PoolType.ReClamm
```

Comment:
we also have some helpers in `pool.helpers.ts` which look very similar

Reply:
added sdk pool type helpers here https://github.com/balancer/frontend-monorepo/pull/1693/commits/ba397b1e20e94991a40a5e945860722e2259d40e